frontend/src/bundles/common/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,27 @@
+import { type ReactNode } from 'react';
+
+import { Navigate } from '~/bundles/common/components/components.js';
+import { AppRoute } from '~/bundles/common/enums/enums.js';
+import { useAppSelector } from '~/bundles/common/hooks/hooks.js';
+import { type AsyncThunkConfig } from '~/bundles/common/types/types.js';
+
+type ProtectedRouteProperties = {
+    children: ReactNode;
+    isPrivate: boolean;
+};
+
+const ProtectedRoute: React.FC<ProtectedRouteProperties> = ({
+    children,
+    isPrivate,
+}) => {
+    const userAuthenticated = useAppSelector(
+        (state: AsyncThunkConfig['state']) => state.auth.user,
+    );
+    if (isPrivate && Object.keys(userAuthenticated).length === 0) {
+        return <Navigate to={AppRoute.SIGN_IN} />;
+    }
+
+    return children;
+};
```

Comment:
```suggestion
    const user = useAppSelector((state) => state.auth.user);
    if (!user) {
      return <Navigate to={AppRoute.SIGN_IN} />;
    }
    return children;
```

Reply:
You don't need isPrivate props for this component. If the route is wrapped, it means that it is private already.